### DISCUSSION TITLE
How to assign a new global variable in CWatM?

### DISCUSSION THREAD

**Discussion by jane-627** [iiasa/CWatM#23] [Q&A] [answered]
Hello, I wonder how to create a new global variable in CWatM and output it, just as the variables after "self.var."? Could you please help me?

**Comment by guillaumot** [comment, marked as answer]
Hi,
yes you just have to create a variable using "self.var", but the variable format should be the same than the other variables (compressed in 1D). Then, you can add this new variable at the end of your settings file.

For example: you can add this line to the groundwater.py module to create the variable "baseflow2":
self.var.baseflow2 = self.var.baseflow * 2
and in the settings file in the [OUTPUT] section : "OUT_MAP_Daily = baseflow2"

Best,
Luca

**Reply by jane-627** [reply]
Thank you very much!

**Reply by jane-627** [reply]
> Hi,
> yes you just have to create a variable using "self.var", but the variable format should be the same than the other variables (compressed in 1D). Then, you can add this new variable at the end of your settings file.
> 
> For example: you can add this line to the groundwater.py module to create the variable "baseflow2":
> self.var.baseflow2 = self.var.baseflow * 2
> and in the settings file in the [OUTPUT] section : "OUT_MAP_Daily = baseflow2"
> 
> Best,
> Luca

Hello, I wonder if I should do more to assign new global variables in CWatM. I've written these code in initial module in waterdemand.py(attached below):
            self.var.irrPaddyRain = globals.inZero.copy()
            self.var.irrNonpaddyRain = globals.inZero.copy()
and these code in dynamic module in waterdemand.py:
            self.var.irrPaddyRain = self.var.fracVegCover[2] * self.var.Rain
            self.var.irrNonpaddyRain = self.var.fracVegCover[3] * self.var.Rain

However, when I tried to run CWatM, Error132 came out:
Error 132: Variable "irrPaddyRain" is not defined in "output_out_map_daily"
Closest variable to this name is: "irrPaddyDemand"

What should I do?
Thank you very much!
Jane
[waterdemand.zip](https://github.com/iiasa/CWatM/files/7115277/waterdemand.zip)

**Reply by guillaumot** [reply]
Hi,
quite curious. It should work like this. Could you add this line just below to check that your new line is active?
self.var.irrPaddyRain = self.var.fracVegCover[2] * self.var.Rain
print('irrPaddyRain is updated here')  # additional line to test your new code

Best,
Luca

**Reply by jane-627** [reply]
> Hi,
> quite curious. It should work like this. Could you add this line just below to check that your new line is active?
> self.var.irrPaddyRain = self.var.fracVegCover[2] * self.var.Rain
> print('irrPaddyRain is updated here') # additional line to test your new code
> 
> Best,
> Luca
Well I've solved the problem. I should change the code in 'water_demand' folder, but not waterdemand.py.
Thank you very much, Luca!
